packages/labs/ssr/src/lib/server-template.ts

```diff
@@ -0,0 +1,67 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {
+  html as baseHtml,
+  type CompiledTemplateResult,
+  type TemplateResult,
+} from 'lit-html';
+
+const SERVER_ONLY = 1;
+
+export interface ServerRenderedTemplate extends TemplateResult {
+  $_litServerRenderMode: typeof SERVER_ONLY;
```

Comment:
If I understand this PR correctly, it is possible to render a `ServerRenderedTemplate` with `lit-html`'s `render` function, since it's a TemplateResult with an extra field that `lit-html` would ignore.

Is this by design so server templates could be shared and rendered as client templates as well (with the caveats that come with client templates)?

If this is by design it may be worth calling out. Technically you could use a `serverhtml` template within a client template if it's only occurring on the client and not during SSR.

If not intended, it may be worth throwing a descriptive error.